plugins/modules/notification/mattermost.py

```diff
@@ -99,12 +123,16 @@ def main():
         argument_spec=dict(
             url=dict(type='str', required=True),
             api_key=dict(type='str', required=True, no_log=True),
-            text=dict(type='str', required=True),
+            text=dict(type='str', default=None),
             channel=dict(type='str', default=None),
             username=dict(type='str', default='Ansible'),
             icon_url=dict(type='str', default='https://www.ansible.com/favicon.ico'),
             validate_certs=dict(default=True, type='bool'),
-        )
+            attachments=dict(type='list', elements='dict'),
+        ),
+        required_one_of=[
+            ('text', 'attachments')
```

Comment:
It's nicer to have a trailing comma, so the next time someone adds another line to this list this line does not need to be modified.
```suggestion
            ('text', 'attachments'),
```